File: studies/movies/ggplot2/bubble_agg.R
library(tidyverse)

df <- read_csv("/Users/nicolas/ets/notascope/data/movies.csv") |>
  group_by(`MPAA Rating`) |>
  summarize(
    `Production Budget` = mean(`Production Budget`, na.rm = TRUE),
    `Worldwide Gross` = mean(`Worldwide Gross`, na.rm = TRUE),
    `IMDB Rating` = mean(`IMDB Rating`, na.rm = TRUE)
  )

ggplot(df) +
  geom_point(aes(
    x = `Production Budget`, y = `Worldwide Gross`,
    color = `MPAA Rating`, size = `IMDB Rating`
  ))

File: studies/movies/plotly_express/bubble_agg.py
import pandas as pd
import plotly.express as px

df = pd.read_csv("data/movies.csv").groupby("MPAA Rating").mean(["Production Budget", "Worldwide Gross", "IMDB Rating"]).reset_index()
fig = px.scatter(df, x="Production Budget", y="Worldwide Gross", color="MPAA Rating", size="IMDB Rating")
fig
